ext/nnstreamer/tensor_decoder/tensordec-boundingbox.h

```diff
@@ -0,0 +1,430 @@
+  protected:
+  guint i_width; /**< Input Video Width */
+  guint i_height; /**< Input Video Height */
+
+  guint max_detection;
+  guint total_labels;
+};
+
+/**
+ * @brief	Class for Bounding box tensor decoder
+ */
+class BoundingBox
+{
+  public:
+  BoundingBox ();
+  ~BoundingBox ();
+
+  gboolean checkLabelProps ();
+  int setBoxDecodingMode (const char *param);
+  int setLabelPath (const char *param);
+  int setVideoSize (const char *param);
+  int setInputModelSize (const char *param);
+  void draw (GstMapInfo *out_info, GArray *results);
+  void logBoxes (GArray *results);
+  void updateCentroids (GArray *boxes);
+
+  int setOption (BoundingBoxOption opNum, const char *param);
+  GstCaps *getOutCaps (const GstTensorsConfig *config);
+  GstFlowReturn decode (const GstTensorsConfig *config,
+      const GstTensorMemory *input, GstBuffer *outbuf);
+
+  private:
+  bounding_box_modes mode;
+  BoxProperties *bdata;
+
+  /* From option2 */
+  imglabel_t labeldata;
+  char *label_path;
+
+  /* From option4 */
+  guint width; /**< Output Video Width */
+  guint height; /**< Output Video Height */
+
+  /* From option6 (track or not) */
+  gint is_track;
+  guint centroids_last_id; /**< The last_id of centroid valid id is 1, 2, ... (not 0). */
+  guint max_centroids_num; /**< The maximum number of centroids */
+  guint consecutive_disappear_threshold; /**< The threshold of consecutive disappeared frames */
+
+  GArray *centroids; /**< Array for centroids */
+  GArray *distanceArray; /**< Array for distances */
+
+  /* From option7 (log or not) */
+  gint do_log;
+
+  gboolean flag_use_label;
+};
+
+/**
+ * @brief	Class for MobilenetSSD box properties
+ */
+class MobilenetSSD : public BoxProperties
+{
+  public:
+  MobilenetSSD ();
+  int mobilenet_ssd_loadBoxPrior ();
+
+  int setOptionInternal (const char *param);
+  int checkCompatible (const GstTensorsConfig *config);
+  GArray *decode (const GstTensorsConfig *config, const GstTensorMemory *input);
+
+  static const int BOX_SIZE = 4;
+  static const int DETECTION_MAX = 2034; /* add ssd_mobilenet v3 support */
+  static const int PARAMS_MAX = 6;
+
+  private:
+  char *box_prior_path; /**< Box Prior file path */
+  gfloat box_priors[BOX_SIZE][DETECTION_MAX + 1]; /** loaded box prior */
+  gfloat params[PARAMS_MAX]; /** Post Processing parameters */
+  gfloat sigmoid_threshold; /** Inverse value of valid detection threshold in sigmoid domain */
+};
+
+/**
+ * @brief	Class for MobilenetSSDPP box properties
+ */
+class MobilenetSSDPP : public BoxProperties
+{
+  public:
+  MobilenetSSDPP ();
+  int get_mobilenet_ssd_pp_tensor_idx (int idx);
+
+  int setOptionInternal (const char *param);
+  int checkCompatible (const GstTensorsConfig *config);
+  GArray *decode (const GstTensorsConfig *config, const GstTensorMemory *input);
+
+  static const guint MAX_TENSORS = 4U;
+
+  private:
+  gint tensor_mapping[MAX_TENSORS]; /* Output tensor index mapping */
+  gfloat threshold; /* Detection threshold */
+};
```

Comment:
With the previous comment satisfied (OCP), you should be able to kick all these derived classes out of this header.

Reply:
Moved derived class to subdir!